binary_search_tree/binary_search_tree.py

```diff
@@ -12,38 +12,164 @@ def __init__(self, value):
 
     # Insert the given value into the tree
     def insert(self, value):
-        pass
+        # Store value of the current tree node, this is going to change with each tree
+        # node so we can't just use self
+        current_tree = self
+        # Declare higher scope variable to hold the value of the next tree node after
+        # we check if the value is smaller or larger
+        next_tree = None
+        if current_tree.value >= value:
+            # Sends it left if the value is smaller
+            next_tree = current_tree.left
+        else:
+            # Sends it right if it's larger
+            next_tree = current_tree.right
+        # While there are still more trees...
+        while next_tree is not None:
+            # Perform a similar action to the initial one above
+            # The value of the current tree is set to the next tree
+            current_tree = next_tree
+            # Then check if the value is smaller or larger than the new current tree
+            if current_tree.value >= value:
+                # Sends it left if the value is smaller
+                next_tree = current_tree.left
+            else:
+                # Sends it right if it's larger
+                next_tree = current_tree.right
+                # Loop breaks once it has gone through every value and hits the right depth location
+
+        # Create a new tree node out of the value using the class constructor
+        new_tree = BinarySearchTree(value)
+        # And then place it on the correct side of the current tree
+        if current_tree.value >= value:
+            current_tree.left = new_tree
+        else:
+            current_tree.right = new_tree
 
     # Return True if the tree contains the value
     # False if it does not
     def contains(self, target):
-        pass
+        # Check if the value of the current tree node is equal to the target value
+        if self.value == target:
+            # If so, return true
+            return True
+        # If not, we go through a similar process as the while loop in the insert method
+        # Store value of the current tree node so that we can mutate it
+        current_tree = self
```

Comment:
this is cool, a reference to the actual raw object, rather than the __self__ of the object